### ISSUE TITLE
Can't type on anything else except English

### ISSUE THREAD

**Issue by cyanoger** [debauchee/barrier#776]
### Operating Systems ###

Server: Windows 10 Pro 64-bit, Version 2004, OS Build 19041.329

Client: Windows 10 Pro Educational 64-bit, Version 1909, OS Build 18363.719

### Barrier Version ###

2.3.2-snapshot-210c2b70
(aka last stable build)

### Steps to reproduce bug ###

Bug number one:
1. Switch to language other than English on host machine (Cyrillic is preferable, not cheked how other languages work)
2. Try to type anything on guest machine
3. Get nothing typed, aka get dead key syndrome.

Bug number two, less ridiculous:
1. Switch to language you want on guest machine
2. Try to type anything on guest machine using it
3. Get symbols typed with host-machine language layout

### Other info ###

* First bug began appearing after some minor windows update on 1909 vesrion of windows on guest machine. Second one appears always.
* The only way to workaround first bug is to type transliteration of cyrillic text on guest or to type in cyrillic on host machine and copy-paste this text on guest machine.
* You can workaround second bug by switching language only on host machine
* I've tried reinstalling Barrier and even using older releases of it.
* These bugs appear even when windows version match on both machines or when host version is lower than guest.
* These bugs don't prevent me from using Barrier entirely.

**Comment by cyanoger** [comment]
https://support.symless.com/hc/en-us/articles/360007325517-International-Keyboard-Restrictions
I guess it might be something related to Synergy bug fixes, and we must wait for them to release everything at synergy core, right?

**Comment by cyanoger** [comment]
2.3.3 - still present

**Comment by shymega** [comment]
> https://support.symless.com/hc/en-us/articles/360007325517-International-Keyboard-Restrictions
> I guess it might be something related to Synergy bug fixes, and we must wait for them to release everything at synergy core, right?

No. We don't generally merge commits from Synergy Core, as we forked from Synergy 2.x - I think, and don't follow their commits generally. We could look into cherry-picking commits from their fix, I suppose, but that would require further discussion: @p12tic - thoughts?

**Comment by shymega** [comment]
> 2.3.3 - still present

Thanks for the update :+1:

**Comment by tdm4** [comment]
Hi,

  I have a similar issue but this time it's between English keyboards.

Server: Mac OS X 10.15.6,  Barrier 2.3.2
Client:  OpenBSD 6.7, Barrier 2.3.2

Both server and client are using British (English) keyboards, but on the Mac (which is the server in this case).. when I type `~` on the Mac, is comes out on the barrier client as `|`. Shift-3 which should be `£` comes out as `#`. So it looks like on the barrier client the keyboard type is set to U.S.

I don't see any option to change this in the barrier server setup on the Mac, nor is there any option in the barrier client options on OpenBSD.

**Comment by darkdragon-001** [comment]
This bug is already apparent in symless/synergy-core#4309 for about five years.

**Comment by JohnnyBoomKek** [comment]
Any fixes or workarounds? It's pretty much impossible to work in a duallingo environment. =(

**Comment by darkdragon-001** [comment]
See meta issue #45 and linked issues.
